File: 백준/Silver/16953. A → B/A → B.py
from collections import deque

A, B = map(int, input().split())
ans = -1
queue = deque([(A, 1)])

while queue:
    a, b = queue.popleft()
    if a == B:
        ans = b
        break
        
    if a * 2 <= B:
        queue.append((a*2, b + 1))
    if int(str(a) + "1") <= B:
        queue.append((int(str(a) + '1'), b + 1))
        
print(ans)

File: 백준/Silver/16953. A → B/A → B.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.on('line', (line) => {
    const [A, B] = line.split(' ').map(Number);
    let ans = -1;
    const queue = [[A, 1]];
    
    while (queue.length > 0) {
        const [a, b] = queue.shift();
        
        if (a === B) {
            ans = b;
            break;
        }
        
        if (a * 2 <= B) {
            queue.push([a * 2, b + 1]);
        }
        
        if (parseInt(a.toString() + '1') <= B) {
            queue.push([parseInt(a.toString() + '1'), b + 1]);
        }
    }
    
    console.log(ans);
    rl.close();
});